lib/bike_brigade/delivery.ex

```diff
@@ -214,34 +214,32 @@ defmodule BikeBrigade.Delivery do
       end
 
     filter =
-      case Keyword.fetch(opts, :start_date) do
-        {:ok, date} ->
-          date_time = LocalizedDateTime.new!(date, ~T[00:00:00])
-          dynamic([campaign: c], ^filter and c.delivery_start >= ^date_time)
-
-        _ ->
-          filter
-      end
-
-    filter =
-      case Keyword.fetch(opts, :end_date) do
-        {:ok, date} ->
-          date_time = LocalizedDateTime.new!(date, ~T[23:59:59])
-          dynamic([campaign: c], ^filter and c.delivery_start <= ^date_time)
-
-        _ ->
-          filter
-      end
-
-    # sometimes we just want to fetch campaigns with specific ids
-    # (such as when we need to display campaigns that urgently need a rider)
-    filter =
+      # First we check if we are fetching with campaign_ids
```

Comment:
I love this but I feel we can refactor this into a `with`

Happy to pair on it!